packages/dds/tree/src/test/forest/jsonCursor.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Jsonable } from "@fluidframework/datastore-definitions";
+import {
+    ITreeCursor,
+    EmptyKey,
+    FieldKey,
+    TreeNavigationResult,
+    TreeType,
+} from "../../..";
+
+/** NodeTypes used by the JsonCursor. */
+export const enum JsonType {
+    Null = 0,
+    Boolean = 1,
+    Number = 2,
+    String = 3,
+    Array = 4,
+    Object = 5,
+}
+
+/**
+ * An ITreeCursor implementation used to read a JSONable tree for testing and benchmarking.
+ */
+export class JsonCursor<T> implements ITreeCursor {
+    // PERF: JsonCursor maintains a stack of nodes/edges traversed.  This stack is
+    //       partitioned across 3 arrays, with the top of the stack stored in fields.
+    //       This design was advantageous in a similar tree visitor, but should
+    //       be measured again to see if this still provides an advantage.
+
+    private currentNode: any;       // The node currently being visited.
```

Comment:
you should be able to have a safer type here (like `unknown`) and/or a more specific type (ex: type Json = Record<string, Json> | number | boolean | null | string | array`).

Reply:
You're right, thx.